### PR TITLE
Support for loggedInToWrite in config

### PR DESCRIPTION
**Pull request by cklokmose**
Added support for the configuration option "loggedInToWrite" that when set to true will require users to be logged in to make any edits to a webstrate.

### CODE REVIEW

File: helpers/PermissionManager.js [hunk 1]
```diff
@@ -239,6 +250,7 @@ module.exports.getUserPermissions = async function(username, provider, webstrate
 
 	let permissions = getCachedPermissions(username, provider, webstrateId);
 	if (permissions) {
+		if (await userMustBeLoggedInToWriteAndUserIsAnonymous(username, provider)) permissions = permissions.replace("w", "");
```
Comment (kbadk) [inline helpers/PermissionManager.js:253]: Although we don't adhere stringently to the rule, we try to keep lines no longer than 100 characters in length. I think readability would benefit from this being on multiple lines.

Also, single quotes over double quotes.

File: helpers/PermissionManager.js [hunk 2]
```diff
@@ -247,6 +259,7 @@ module.exports.getUserPermissions = async function(username, provider, webstrate
 		snapshot);
 
 	setCachedPermissions(username, provider, permissions, snapshot.id);
+	if (await userMustBeLoggedInToWriteAndUserIsAnonymous(username, provider)) permissions = permissions.replace("w", "");
```
Comment (kbadk) [inline helpers/PermissionManager.js:262]: Same as above.

File: helpers/PermissionManager.js [hunk 3]
```diff
@@ -223,6 +223,17 @@ module.exports.expireAllAccessTokens = function(webstrateId, local) {
 	}
 };
 
+/**
+ * Check if configuration option is set so users must be logged in to write in a webstrate
+ * and the given user is anonymous
```
Comment (kbadk) [inline helpers/PermissionManager.js:228]: Very nitpicky, but the other comments usually end with a period. 😬

File: helpers/PermissionManager.js [hunk 4]
```diff
@@ -223,6 +223,17 @@ module.exports.expireAllAccessTokens = function(webstrateId, local) {
...
+ * @param  {string}   username    Username.
+ * @param  {string}   provider    Login provider (GitHub, Facebook, OAuth, ...).
+ * @return {bool}                 The configuration option is set and the user is anonymous
```
Comment (kbadk) [inline helpers/PermissionManager.js:231]: Same as above.

File: helpers/PermissionManager.js [hunk 5]
```diff
@@ -223,6 +223,17 @@ module.exports.expireAllAccessTokens = function(webstrateId, local) {
...
+ */
```
Comment (kbadk) [inline helpers/PermissionManager.js:232]: Right about here, we tend to include either a `@private` or `@public` flag. This function isn't exported, so it'd be `@private`. But I see we aren't very consistent about this throughout the codebase, so feel free to ignore.

**Review by kbadk — CHANGES_REQUESTED** [review]

File: helpers/PermissionManager.js (hunk 2 above)
Comment (kbadk) [reply helpers/PermissionManager.js:262]: Also, consider that if the `permissions` string was accidentally (or maliciously) set to `rww`, `permissions.replace("w", "");` would only remove the first occurrence, i.e. `permissions` would afterwards be `rw`, and the webstrate thus still annonymously writeable, regardless of the `loggedInToWrite` to write flag.

This potential problem could be avoided by doing `permissions.replace(/w/g, '')`.

File: helpers/PermissionManager.js (hunk 1 above)
Comment (cklokmose) [reply helpers/PermissionManager.js]: I split it up in multiple lines

File: helpers/PermissionManager.js (hunk 2 above)
Comment (cklokmose) [reply helpers/PermissionManager.js]: I changed it to `permissions.replace(/w/g, '')`

**Review by kbadk — APPROVED** [review]

**Pull request merged**